torch/_functorch/_aot_autograd/schemas.py

```diff
@@ -481,6 +481,12 @@ def __post_init__(self):
         # this information.
         self.num_forward = self.num_forward_returns + self.num_outputs_rng_offset
 
+        # Indexes of saved tensors which are donated buffer.
+        # Donated buffer means the tensor is not alias of any forward user input, forward user output,
+        # and backward output.
+        # The index ranges from 0 to # saved tensors.
+        self.fw_donated_buffer: Optional[List[int]] = None
```

Comment:
cc @jamesjwu, new metadata. For this one though, we actually only really need it to be part of the cache key for the backward graph (it impacts how inductor compiles the backward, but is irrelevant to the fw)

Reply:
Actually as long as we’re ever compiling the backward ahead of time I guess this will need to be part of the cache key too